app/src/main/java/com/sdp/movemeet/view/map/GPSRecordingActivity.java

```diff
@@ -70,15 +94,21 @@ public void onLocationResult(@NonNull LocationResult locationResult) {
         }
     };
 
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    public ArrayList<LatLng> path;
-
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_g_p_s_recording);
 
+        Intent intent = getIntent();
+        activity = (Activity) intent.getSerializableExtra(ActivityDescriptionActivity.RECORDING_EXTRA_NAME);
+
+        if (AuthenticationInstanceProvider.getAuthenticationInstance().getCurrentUser() == null || activity == null) {
+            intent = new Intent(GPSRecordingActivity.this, MainActivity.class);
```

Comment:
If the `activity` is `null`, I agree that we have to redirect the user to the `MainActivity`, but if the "Firebase User" is `null`, shouldn't we rather redirect the user directly to the `LoginActivity`?